src/components/production-block.tsx

```diff
@@ -61,31 +61,38 @@ export const ProductionBlock = (props: ProductionBlockProps) => {
   useEffect(() => {
     const onMessage = async (event: MessageEvent) => {
       if (event.data.id === id.current) {
-        const { data, origin } = event;
+        const { data, origin, source } = event;
 
         // handle messages from the sandboxed block
         const originRegex = new RegExp(
           /^https:\/\/\d{1,4}-\d{1,4}-\d{1,4}-sandpack\.codesandbox\.io$/
         );
-        if (!originRegex.test(origin)) return;
+        if (!source || !originRegex.test(origin)) return;
         if (data.type === "github-data--request") {
-          const res = await onRequestGitHubData(
-            data.path,
-            data.params,
-            data.id
-          );
-          const iframe = sandpackWrapper.current?.querySelector("iframe");
-          if (!iframe) return;
-          iframe.contentWindow?.postMessage(
-            {
-              type: "github-data--response",
-              id: id.current,
-              data: res,
-            },
-            "*"
-          );
+          onRequestGitHubData(data.path, data.params)
+            .then((res) => {
+              source.postMessage(
+                {
+                  type: "github-data--response",
+                  id: id.current,
+                  data: res,
+                },
+                origin as any
```

Comment:
`postMessage` has an overloaded type and for some reason TypeScript won't resolve it properly

Reply:
this will always be a string, right?

Reply:
yeah, `event.origin` is typed as a string.

The problem is that `event.source` can be any of `Window`, `ServiceWorker`, or `MessagePort`, and only `Window` supports this form of `postMessage`; I fixed it with a cast to `Window`.